Controllers/StandUpController.js

```diff
@@ -99,171 +93,70 @@ class StandUpController {
     }
 
     try {
-      let standUp = await StandUp.findById(req.params.id);
-      if (!standUp) {
-        throw new Error(`StandUp of id '${req.params.id}' not found!`);
-      }
-
-      const standupBody = req.body.standup;
+      let { standUp } = req;
+      const { body } = req;
 
-      if (standupBody.reminders) {
-        let standUpSchedules = standUp.reminders.schedules;
-        const newSchedules = [];
-        const updateSchedules = [];
-
-        if (!standupBody.reminders.days) {
-          standupBody.reminders.days = standUp.reminders.days;
-        }
+      if (body.reminders) {
+        const standUpSchedules = standUp.reminders.schedules;
 
         // checks of schedule(s) option
-        if (standupBody.reminders.schedules) {
-          standupBody.reminders.schedules.forEach((schedule) => {
-            if (String(Object.keys(schedule)) === String(['time'])) {
-              newSchedules.push(schedule);
-              if (newSchedules.length + standUpSchedules.length > 3) {
-                throw new Error(
-                  `There is currently ${standUpSchedules.length} existing reminders and only up to 3 can be scheduled in a standup!`,
-                );
-              }
-            } else if (
-              String(Object.keys(schedule)) === String(['_id', 'time'])
-            ) {
-              updateSchedules.push(schedule);
-            } else {
-              throw new Error(`${Object.keys(schedule)} is not valid!`);
-            }
-          });
+        if (body.reminders.schedules && body.reminders.schedules.length !== 0) {
+          const users = await User.find({ standups: standUp._id })
+            .select('_id')
+            .lean();
+          const userIds = users.map((user) => user._id);
 
-          if (updateSchedules.length !== 0) {
-            const scheduleIds = updateSchedules.map((schedule) => schedule._id);
-            const undefinedSchedules = StandUpHelpers.remindersArrDiff(
-              standUpSchedules,
-              scheduleIds,
+          const unifiedSchedules = body.reminders.schedules.reduce((acc, schedule) => {
```

Comment:
hello @Fenn-CS just want to explain a little about this part;
## In the case if the user wants to update the standup that has reminder schedules
- This part walks through each of the reminder schedules users have specified
- Steps:
   1. It checks if the schedule to be updated already exists in the to-be updated standup document
   2. If (i) is true, it will move on to check on the notification time of that existing schedule
     - The reason why this needs to be done is that:
       - No time zone/offset from UTC is saved along with the standup
       - The current vision I have is that in the standupman UI, the user can set the time zone they want to follow to send out reminders. We can detect the user time zone through the client browser and set it as a default value for them. However, if they wish to change the time zone to other rather than the defaulted one, they can do so.
       - Therefore, the check is needed to see if the notification time generated differs from the currently existing notification time. As the notification time is generated based on the time zone then converted to UTC.
       - If the notification time is different, it will overwrite with the new generated date time. This reminder schedule is set to be sent the next day.
         - If a user updates any reminder schedules or creates a new reminder schedule today, the created/updated schedule will only occur the next day.
         - This ensures that duplicated reminders will not be sent out.
         - For instance, if the current time is 2 p.m. and a standup reminder has already been sent out to users at 10 a.m. If the user updates the schedule again, less than the current time (2 p.m.) at 11 a.m., and if a day is not added to the date time, the job will detect it and send it out again.
         - Or should we have the consideration to see if the updated schedule is less/more than the current time? For instance, the current time is 2 p.m. if user updates a reminder schedule to 3 p.m. it will still be counted to be sent out today. However, if the time has already passed e.g. user updates the reminders to 11 a.m. it will be counted to be sent out the next day.
       - Otherwise, it will just return the existing notification time.
    3. If (i) is not true, it will mean that it is a new reminder that the user wishes to generate. It will go ahead to generate the new reminder and it will only occur the next day.